pkg/apis/multiclusterservice/v1/types.go

```diff
@@ -0,0 +1,56 @@
+package v1
+
+import (
+	"net"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
```

Comment:
These special comments are directives for the k8s code generator. Do you plan to generate all the Clientset code as well?

Reply:
Yes.  I think that will be required to use this CRD, is that correct? Also one of the file zz_generated.deepcopy.go  is auto-generated.

Reply:
It's not required but the generated code makes it much easier to use. You can use [this script wrapper](https://github.com/submariner-io/submariner/blob/master/scripts/codegen) to generate the code.

Reply:
I will push the generated client set code as a separate patch.

Reply:
Added the client generated code.